homeassistant/components/netatmo/__init__.py

```diff
@@ -79,6 +88,30 @@
             hass.config_entries.async_forward_entry_setup(entry, component)
         )
 
+    if CONF_WEBHOOK_ID not in entry.data:
+        entry.data[CONF_WEBHOOK_ID] = secrets.token_hex()
```

Comment:
Don't write directly to entry.data. Make a copy.
```suggestion
        data = {**entry.data, CONF_WEBHOOK_ID: secrets.token_hex()}
```